### PR TITLE
:bug: Actually rename jsx files to tsx files

### LINKED ISSUE #144: Converting JSX with --delete-source argument leads to file being deleted
As touched upon in #138, files with a `jsx` extension are not renamed when converting using the `--write` argument. A consequence is that the source file is overwritten with the conversion results. I have discovered another consequence: If you supply the `--delete-source` argument, the result file (which, to repeat myself, has the same name as the source file) is deleted. 

I have tried this on a large project with about 400 JSX files of varying complexity, each one disappearing. But it happens even when the file is empty.

```zsh
mkdir empty-dir
cd empty-dir
touch foo.js bar.jsx
npx -p @khanacademy/flow-to-ts flow-to-ts --write --delete-source foo.js bar.jsx
ls # foo.ts
```

The console output for .jsx files is completely fine. And the file contents are all right when you use --write (except that you would expect the result file to have the .tsx extension). 

I am guessing that the underlying problem is an assumption that the extension should have changed from jsx to tsx.

### PR DESCRIPTION
**Pull request by Vages**
Fixes #144 and #138

### CODE REVIEW

**Comment by Vages** [comment]
[clabot:check]

**Comment by benbrandt** [comment]
big +1 to this!

**Comment by Vages** [comment]
But if you want to use this patch, @benbrandt, you could add the following line to your `devDependencies` in `package.json`: `"@khanacademy/flow-to-ts": "https://github.com/Vages/flow-to-ts.git#325d49b8167ec714855f17ce4a8852684cc4d7a2",`. Works with yarn, at least (don't know about `npm`).

**Comment by k4b7** [comment]
@Vages thanks for the PR.  The fix looks good.  Do you mind adding a test to [test/cli.test.js](https://github.com/Khan/flow-to-ts/blob/master/test/cli.test.js)?

**Comment by Vages** [comment]
It is added now, @kevinbarabash!

File: test/cli.test.js [hunk 1]
```diff
@@ -172,6 +172,45 @@ describe("cli", () => {
...
+
+    // Assert
+    expect(fs.existsSync(path.join(tmpdir, "foo.ts"))).toBe(true);
+    expect(fs.existsSync(path.join(tmpdir, "bar.tsx"))).toBe(true);
+    expect(
+      fs.existsSync(path.join(tmpdir, "baz.ts")) // Uses .ts extension if no JSX syntax found
+    ).toBe(true);
+    expect(fs.existsSync(path.join(tmpdir, "foo.jsx"))).toBe(false);
+    expect(fs.existsSync(path.join(tmpdir, "bar.jsx"))).toBe(false);
+    expect(fs.existsSync(path.join(tmpdir, "bar.jsx"))).toBe(false);
```
Comment (k4b7) [inline test/cli.test.js]: I think the last `expect` here should be for `baz.jsx`.

**Review by k4b7 — APPROVED** [review]
LGTM.  Thanks for adding a test.  There's a minor issue with the test that I should be able to patch.

**Comment by k4b7** [comment]
I've published 0.1.6 and it contains this change.

File: test/cli.test.js (hunk 1 above)
Comment (Vages) [reply test/cli.test.js]: Late reply, but yes. 👍

**Pull request merged**